src/test/java/io/thinkit/edc/client/connector/ApplicationObservabilityTest.java

```diff
@@ -38,139 +38,65 @@ void should_check_health() {
-                            assertThat(failure.failureDetail()).isNotNull().isEqualTo("string");
-                            assertThat(failure.messages().size()).isGreaterThan(0);
-                        });
-                    });
-        } catch (InterruptedException | ExecutionException e) {
-            throw new RuntimeException(e);
+        @Test
+        void should_check_readiness() {
+            var result = applicationObservability.checkReadiness();
+            health_status_response(result);
         }
-    }
 
-    @Test
-    void should_check_startup() {
-        var result = applicationObservability.checkStartup();
+        @Test
+        void should_check_startup() {
+            var result = applicationObservability.checkStartup();
+            health_status_response(result);
+        }
 
-        assertThat(result.isSucceeded()).isTrue();
-        assertThat(result.getContent().isSystemHealthy()).isTrue();
-        assertThat(result.getContent().componentResults()).isNotNull().first().satisfies(componentResult -> {
-            assertThat(componentResult.component()).isNotNull().isEqualTo("string");
-            assertThat(componentResult.isHealthy()).isTrue();
-            assertThat(componentResult.failure()).isNotNull().satisfies(failure -> {
-                assertThat(failure.failureDetail()).isNotNull().isEqualTo("string");
-                assertThat(failure.messages().size()).isGreaterThan(0);
-            });
-        });
+        @Test
+        void should_check_liveness() {
+            var result = applicationObservability.checkLiveness();
+            health_status_response(result);
+        }
     }
 
-    @Test
-    void should_check_startup_async() {
-        try {
-            var result = applicationObservability.checkStartupAsync().get();
-            assertThat(result.isSucceeded()).isTrue();
-            assertThat(result.getContent().isSystemHealthy()).isTrue();
-            assertThat(result.getContent().componentResults())
-                    .isNotNull()
-                    .first()
-                    .satisfies(componentResult -> {
-                        assertThat(componentResult.component()).isNotNull().isEqualTo("string");
-                        assertThat(componentResult.isHealthy()).isTrue();
-                        assertThat(componentResult.failure()).isNotNull().satisfies(failure -> {
-                            assertThat(failure.failureDetail()).isNotNull().isEqualTo("string");
-                            assertThat(failure.messages().size()).isGreaterThan(0);
-                        });
-                    });
-        } catch (InterruptedException | ExecutionException e) {
-            throw new RuntimeException(e);
+    @Nested
+    class Async {
+        @Test
+        void should_check_health_async() {
+            var result = applicationObservability.checkHealthAsync();
+            assertThat(result)
+                    .succeedsWithin(5, TimeUnit.SECONDS)
```

Comment:
a `Duration` field representing the async test timeout could be kept in the `ContainerTestBase` class, just to avoid repeating this everywhere and to have a single point where it could be changed then